spec/services/courses/query_spec.rb

```diff
@@ -847,6 +859,52 @@ def initialize(course, minimum_distance_to_search_location:)
           ]
         end
       end
+
+      context 'when latitude and longitude are given' do
+        it 'always order by distance and ignores the order param' do
+          %w[
+            course_name_ascending
+            course_name_descending
+            provider_name_ascending
+            provider_name_descending
+          ].each do |order|
+            results = described_class.call(
+              params: {
+                latitude: london.latitude,
+                longitude: london.longitude,
+                radius: 10,
+                order:
+              }
+            )
+            expect(results).to match_collection(
+              [
+                course_london_result,
+                course_canary_wharf_result,
+                course_lewisham_result
+              ],
+              attribute_names: %w[name minimum_distance_to_search_location]
+            )
+          end
+        end
```

Comment:
This is important to have this test because on the location search the SQL needs to group the courses but the custom order needs to add the `SELECT provider.provider_name` which will cause SQL errors.  The location search does not need to include this select since is ordering by location.